### PR TITLE
AGAT - Add new feature (functional annotation)

### PR DESCRIPTION
**Pull request by rlibouba**
Hello, 
I've opened this PR to add the function annotation to this tool. 
I'm having a problem defining filters for the outputs of this new function. 
Two output files are generated: report.txt and error.txt
The user can add a blast output with the database to complete the functionality. By doing so, another output is generated: duplicatedNameFromBlast.txt
You can also do the same thing by adding an interpro output where 5 other outputs will be generated. 
The user can use the blast and interpro outputs simultaneously or separately. 

Even when there is no blast or interpro output. 
Can you help me?

### CODE REVIEW

File: tools/agat/agat.xml [hunk 1]
```diff
@@ -96,7 +96,7 @@
             agat_sq_filter_feature_from_fasta.pl 
                 --gff $input_annotation 
                 --fasta $ref_genome 
-                --config $agat_configfile
+                --config $wYIG·agat_configfile
```
Comment (bgruening) [inline tools/agat/agat.xml:99]: that looks a bit wired ...

File: tools/agat/agat.xml [hunk 2]
```diff
@@ -115,6 +115,23 @@
                 --config $agat_configfile
                 --output 'output' &&
             cat 'output' > '${annotation}'
+        #else if $tool.selector == 'functional_annotation'
+        @input_annotation_single@
```
Comment (bgruening) [inline tools/agat/agat.xml:119]: indentation

File: tools/agat/agat.xml [hunk 3]
```diff
@@ -115,6 +115,23 @@
...
+            agat_sp_manage_functional_annotation.pl
+                --gff $input_annotation
```
Comment (bgruening) [inline tools/agat/agat.xml:121]: I guess this one needs to be quoted also above.

File: tools/agat/agat.xml [hunk 4]
```diff
@@ -302,6 +356,30 @@ check_identical_isoforms: true
...
             <filter>tool['selector'] == 'annotation_statistics'</filter>
         </collection>
+        <data name="sequence_annot_functionnal_report" format="txt" from_work_dir="galaxy_functional_annot/report.txt" label="${tool.name} on ${on_string}: Report">
+            <filter>tool['selector'] =='functional_annotation'</filter>
+        </data>
+        <data name="sequence_annot_functionnal_error" format="txt" from_work_dir="galaxy_functional_annot/error.txt" label="${tool.name} on ${on_string}: Error">
+            <filter>tool['selector'] =='functional_annotation'</filter>
+        </data>
+        <data name="sequence_annot_functionnal_duplicate_blast" format="txt" from_work_dir="galaxy_functional_annot/duplicatedNameFromBlast.txt" label="${tool.name} on ${on_string}: Duplicate name from Blast">
+            <filter>tool['selector'] =='functional_annotation' and blast_adv == 'yes'</filter>
```
Comment (bgruening) [inline tools/agat/agat.xml:366]:
```suggestion
            <filter>tool['selector'] =='functional_annotation' and blast_adv['options'] == 'yes'</filter>
```

Comment (bgruening) [reply tools/agat/agat.xml:366]: have you tried this?

Comment (rlibouba) [reply tools/agat/agat.xml:366]: Hello, thank you for your feedback.
I tried that, but it doesn't work.

File: tools/agat/agat.xml [hunk 5]
```diff
@@ -273,6 +292,41 @@ check_identical_isoforms: true
                 <expand macro="ANNOTATION_INPUT" format="gff,gff3,gff3.gz"/>
                 <expand macro="AGAT_CONFIG"/>
             </when>
+
+            <when value="functional_annotation">
+                <expand macro="ANNOTATION_INPUT" format="gff,gff3,gff3.gz"/>
+                <conditional name="interpro_adv">
```
Comment (abretaud) [inline tools/agat/agat.xml:298]: No need for a conditional here, just an optional input type=data

File: tools/agat/agat.xml [hunk 6]
```diff
@@ -273,6 +292,41 @@ check_identical_isoforms: true
...
+                        <param argument="-b" name="input_blast" type="data" format="tabular" optional="true" label="Input blast file"/>
+                        <param argument="--db" name="database" type="data" format="fasta" optional="true" label="Fasta file using for the blast" help="Gene names and products/descriptions will be fished from this file." />
+                        <param argument="--blast_evalue" name="evalue" type="float" min="0" max="1" value="0.000001" label="Maximum e-value to keep the annotation from the blast file"/>
+                    </when>
+                    <when value="no">
+                    </when>
+                </conditional>
+                <param name="prot_existence" type="select" label="Protein existence (PE)" help="You can decide until which protein existence level you want to consider to lift the finctional information">
+                    <option value="1">Experimental evidence at protein level</option>
+                    <option value="2"> Experimental evidence at transcript level</option>
```
Comment (abretaud) [inline tools/agat/agat.xml:324]: unneeded space

File: tools/agat/agat.xml (hunk 4 above)
Comment (abretaud) [reply tools/agat/agat.xml:366]: I guess it should be `tool['blast_adv']['options']`

**Comment by bgruening** [comment]
@rlibouba only two errors left.

One is this here:

> @@ -0,0 +1,3 @@
+ERROR AP010953.1 not found among the db! You probably didn't give to me the same fasta file than the one used for the blast. (l2=nz_cp027599.1)
+We have 1 description candidates.
+error nothing found for AP010953.1-nz_cp027599.1

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
Thanks a lot @rlibouba!

**Comment by bgruening** [comment]
Will be updated automatically over the weekend.

**Pull request merged**